src/main/kotlin/ar/edu/unahur/obj2/caralibro/Usuario.kt

```diff
@@ -2,10 +2,85 @@ package ar.edu.unahur.obj2.caralibro
 
 class Usuario {
   val publicaciones = mutableListOf<Publicacion>()
+  val publicacionesMeGustan = mutableListOf<Publicacion>()
+  val amigos = mutableListOf<Usuario>()
 
   fun agregarPublicacion(publicacion: Publicacion) {
     publicaciones.add(publicacion)
   }
 
   fun espacioDePublicaciones() = publicaciones.sumBy { it.espacioQueOcupa() }
-}
+
+  fun darMeGusta(publicacion: Publicacion) {
+    if (!publicacionesMeGustan.contains(publicacion)) {
+      publicacion.meGusta()
+      publicacion.agregarUsuario(this)
+      publicacionesMeGustan.add(publicacion)
+    }
+  }
+
+  fun totalMeGusta() = publicaciones.sumBy { it.cantidadMeGusta }
+
+  fun agregarAmigo(usuario: Usuario) {
+    if (!amigos.contains(usuario)) {
+      amigos.add(usuario)
+    }
+  }
+
+  fun totalAmigos() = this.amigos.size
+
+  fun esAmigoDe(usuario: Usuario) = amigos.contains(usuario)
+
+  fun soyMasAmistosoQue(usuario: Usuario) = this.totalAmigos() > usuario.totalAmigos()
+
+  fun puedeVerPublicacion(usuario: Usuario, publicacion: Publicacion): Boolean {
+    var resultado = true
+    if (publicacion.permiso == "amigos" && !usuario.esAmigoDe(this)) {
+      resultado = false
+    }
+      else if (((publicacion.permiso == "privado") || (publicacion.permiso == "excluidos"))
+        && this.estaExcluidoYNoPermitido(publicacion)) {
+       resultado = false
+      }
+      else if (!usuario.esAmigoDe(this) && publicacion.permiso == "privado") {
+        resultado = false
+      }
+    return resultado
+  }
+
+  fun estaExcluidoYNoPermitido(publicacion: Publicacion) = publicacion.excluidos.contains(this) &&
+          !publicacion.permitidos.contains(this)
+
+  fun mejoresAmigos() = amigos.filter { it.puedeVerTodasLasPublicacionesDe(this) }
+
+  fun puedeVerTodasLasPublicacionesDe(usuario: Usuario): Boolean {
+    var resultado = true
+    if (usuario.publicaciones.any { p -> !this.puedeVerPublicacion(usuario, p) }) resultado = false
+    return resultado
+  }
```

Comment:
El método `any` de las listas ya devuelve un booleano, es innecesario guardarlo en una variable.
```suggestion
  fun puedeVerTodasLasPublicacionesDe(usuario: Usuario) = usuario.publicaciones.any { p -> !this.puedeVerPublicacion(usuario, p) })
``` 
Prácticamente un método así es cómo preguntar ¿Hay alguno que puede ver todas las publicaciones de X? En dónde te van a responder sí o no. Les aconsejo le peguen una leida a este [apunte](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-list/) o [este](https://devhints.io/kotlin)